### PR TITLE
Fix Istio ingress service.yaml and add test.

### PR DESCRIPTION
**Pull request by axel7born**
**How to categorize this PR?**

/area networking
/kind bug

**What this PR does / why we need it**:
This PR fixes the the Istio ingress service template for the DualStack case.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```bugfix developer
Fix Istio ingress service.yaml for dual-stack setup and add test.
```

### CODE REVIEW

File: pkg/component/istio/istio_test.go [hunk 1]
```diff
@@ -528,10 +539,13 @@ var _ = Describe("istiod", func() {
...
 			It("should successfully deploy correct external traffic policy", func() {
 				Expect(c.Get(ctx, client.ObjectKeyFromObject(managedResourceIstioSecret), managedResourceIstioSecret)).To(Succeed())
-				Expect(diffConfig(string(managedResourceIstioSecret.Data["istio-ingress_templates_service_test-ingress.yaml"]), istioIngressService(&externalTrafficPolicy))).To(BeEmpty())
+				Expect(diffConfig(string(managedResourceIstioSecret.Data["istio-ingress_templates_service_test-ingress.yaml"]), istioIngressServiceETPCluster())).To(BeEmpty())
 			})
 		})
 
+
+
+
```
Comment (ScheererJ) [inline pkg/component/istio/istio_test.go]: Can you please remove two whitespace lines? One empty line should be sufficient.

File: pkg/component/istio/istio_test.go [hunk 2]
```diff
@@ -554,7 +568,33 @@ var _ = Describe("istiod", func() {
 
 			It("should successfully deploy correct external traffic policy", func() {
 				Expect(c.Get(ctx, client.ObjectKeyFromObject(managedResourceIstioSecret), managedResourceIstioSecret)).To(Succeed())
-				Expect(diffConfig(string(managedResourceIstioSecret.Data["istio-ingress_templates_service_test-ingress.yaml"]), istioIngressService(&externalTrafficPolicy))).To(BeEmpty())
+				Expect(diffConfig(string(managedResourceIstioSecret.Data["istio-ingress_templates_service_test-ingress.yaml"]), istioIngressServiceETPLocal())).To(BeEmpty())
+			})
+		})
+
+
```
Comment (ScheererJ) [inline pkg/component/istio/istio_test.go]: Please just use a single empty line between tests/contexts.

File: pkg/component/istio/charts/istio/istio-ingress/templates/service.yaml [hunk 3]
```diff
@@ -27,8 +27,8 @@ spec:
 {{- end }}
 {{- if .Values.externalTrafficPolicy }}
   externalTrafficPolicy: {{ .Values.externalTrafficPolicy -}}
-{{- end -}}
-{{- if eq .Values.dualStack true -}}
+{{- end }}
+{{- if eq .Values.dualStack true }}
```
Comment (ScheererJ) [inline pkg/component/istio/charts/istio/istio-ingress/templates/service.yaml:31]: Why do we have a trailing white space removal in the `externalTrafficPolicy` value? Compared to other values, e.g. type, version or load balancer IP this looks weird.

**Review by ScheererJ — COMMENTED** [review]
Thanks a lot for the pull request.

I have two minor nits and a question, though.

**Comment by ScheererJ** [comment]
@axel7born Please check the unit tests and run `make generate`.

File: pkg/component/istio/charts/istio/istio-ingress/templates/service.yaml (hunk 3 above)
Comment (axel7born) [reply pkg/component/istio/charts/istio/istio-ingress/templates/service.yaml:31]: I removed it.

**Review by ScheererJ — APPROVED** [review]
/lgtm
/approve

**Pull request merged**